bundles/org.openhab.binding.gce/src/main/java/org/openhab/binding/gce/model/StatusFileInterpreter.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.gce.model;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.gce.internal.handler.Ipx800EventListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * This class takes care of interpreting the status.xml file
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class StatusFileInterpreter {
+    private static final String URL_TEMPLATE = "http://%s/globalstatus.xml";
+    private final Logger logger = LoggerFactory.getLogger(StatusFileInterpreter.class);
+    private final String hostname;
+    private @Nullable Document doc;
+    private final Ipx800EventListener listener;
+
+    public static enum StatusEntry {
+        VERSION,
+        CONFIG_MAC;
+    }
+
+    public StatusFileInterpreter(String hostname, Ipx800EventListener listener) {
+        this.hostname = hostname;
+        this.listener = listener;
+    }
+
+    public void read() {
+        try {
+            DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+            URL url = new URL(String.format(URL_TEMPLATE, hostname));
+
+            URLConnection conn = url.openConnection();
+            Document document = builder.parse(conn.getInputStream());
+            document.getDocumentElement().normalize();
+            doc = document;
+            pushDatas();
+        } catch (IOException | SAXException | ParserConfigurationException e) {
```

Comment:
You need to close the connection, when you are finished by closing the input stream.